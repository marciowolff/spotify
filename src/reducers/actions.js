export { default as actionsSpotify } from './spotify/actions';
