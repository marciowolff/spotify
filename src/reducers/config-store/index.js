import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import rootReducer from '../';

export default ({ initialStates } = {}) => {
  const enhancer = compose(
    applyMiddleware(thunk),
    logger()
  );

  return createStore(rootReducer, initialStates, enhancer);
};

const logger = () =>
  window.__REDUX_DEVTOOLS_EXTENSION__
    ? window.__REDUX_DEVTOOLS_EXTENSION__()
    : x => x;
