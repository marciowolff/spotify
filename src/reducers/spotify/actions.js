export const SPOTIFY_LIST = 'spotify:SPOTIFY_LIST';
export const SPOTIFY_ALBUM = 'spotify:SPOTIFY_ALBUM';

const setList = (albums = []) => dispatch => {
  dispatch({
    type: SPOTIFY_LIST,
    payload: albums,
  });
};

const setAlbum = (content = {}) => dispatch => {
  dispatch({
    type: SPOTIFY_ALBUM,
    payload: content,
  });
};

export default {
  setList,
  setAlbum,
};
