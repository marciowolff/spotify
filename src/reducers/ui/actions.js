export const ALERT_MESSAGE = 'ui:ALERT_MESSAGE';
export const LOADING = 'ui.LOADING';

export const alertMessage = (message = '') => dispatch => {
  dispatch({
    type: ALERT_MESSAGE,
    payload: {
      message,
    },
  });
};

export const setLoading = (loading = false) => dispatch => {
  dispatch({
    type: LOADING,
    payload: {
      loading,
    },
  });
};
