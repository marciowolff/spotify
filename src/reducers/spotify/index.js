import createReducer from '../create-reducer';

import { SPOTIFY_LIST, SPOTIFY_ALBUM } from './actions';

export const initialState = {};

const spotify = createReducer(initialState, {
  [SPOTIFY_LIST]: (state, action) => ({
    ...state,
    list: action.payload,
  }),
  [SPOTIFY_ALBUM]: (state, action) => ({
    ...state,
    album: action.payload,
  }),
});

export default spotify;
