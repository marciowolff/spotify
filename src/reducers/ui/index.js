import createReducer from '../create-reducer';
import { ALERT_MESSAGE, LOADING } from './actions';

export const initialState = {
  alertMessage: '',
  loading: true,
};

const ui = createReducer(initialState, {
  [ALERT_MESSAGE]: (state, action) => ({
    ...state,
    alertMessage: action.payload.message,
  }),
  [LOADING]: (state, action) => ({
    ...state,
    loading: action.payload.loading,
  }),
});

export default ui;
