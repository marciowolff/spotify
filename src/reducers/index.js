import { combineReducers } from 'redux';
import ui from './ui';
import spotify from './spotify';

export default combineReducers({
  ui,
  spotify,
});
