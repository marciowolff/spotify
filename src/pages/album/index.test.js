import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import { actionsSpotify } from '../../reducers/actions';

import Album from './index';

const mockDispatch = jest.fn();
jest.mock('react-redux', () => ({
  useDispatch: () => mockDispatch,
  useSelector: () => ({ spotify: {} }),
}));

jest.mock('../@container', () => ({ children }) => (
  <div className="container">{children}</div>
));

const history = { push: jest.fn() };
const location = { pathname: '/teste' };
const match = { params: 'codParam' };
const mockAlbums = jest.fn();

const spysetAlbum = jest.spyOn(actionsSpotify, 'setAlbum');

let wrapper;

beforeEach(() => {
  mockAlbums.mockImplementation(() => Promise.resolve({ items: {} }));
  wrapper = mount(
    <Album
      location={location}
      history={history}
      match={match}
      service={{ spotify: { albums: mockAlbums } }}
    />
  );
});

afterEach(() => {
  spysetAlbum.mockClear();
});

it('Album should be a function', () => {
  expect(typeof Album).toBe('function');
});
it('Should Album to match snapshot', () => {
  const tree = renderer
    .create(<Album location={location} history={history} match={match} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

describe('render page default', () => {
  it('Should render component Container', () => {
    const CpContainer = wrapper.find('.container');
    expect(CpContainer).toHaveLength(1);
  });

  it('Should render button callback', () => {
    expect(wrapper.find('Button')).toHaveLength(1);
  });

  it('Should render component Thumb', () => {
    const CpThumb = wrapper.find('Thumb');
    expect(CpThumb).toHaveLength(1);
    expect(CpThumb.props().data).toEqual({ spotify: {} });
  });

  it('Should render component List', () => {
    const CpList = wrapper.find('List');
    expect(CpList).toHaveLength(1);
    expect(CpList.props().data).toEqual({ spotify: {} });
  });
});

describe('Events', () => {
  it('Should button callback', () => {
    wrapper.find('Button').simulate('click');

    expect(history.push).toHaveBeenCalledWith('/home');
  });
});
