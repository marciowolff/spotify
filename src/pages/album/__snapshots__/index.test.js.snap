// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Album to match snapshot 1`] = `
<div
  className="container"
>
  <a
    className="button "
    onClick={[Function]}
    type="link"
  >
    &lt; Voltar
  </a>
  <section>
    <aside
      className="thumb album-thumb"
    >
      <figure
        className="media "
      >
        <img
          alt=" - "
          src="mock-file"
        />
        <figcaption>
          <span>
             
          </span>
        </figcaption>
      </figure>
    </aside>
    <ul
      className="list"
    />
  </section>
</div>
`;
